File: src/main/java/calculadora/command/AddCommand.java
package calculadora.command;

/**
 *
 * @author vedz11
 */
import calculadora.memento.*;

public class AddCommand implements Command {
    private final Calculator calculator;
    private final double value;
    private CalculatorMemento previousState;

    public AddCommand(Calculator calculator, double value) {
        this.calculator = calculator;
        this.value = value;
    }

    @Override
    public void execute() {
        previousState = calculator.save();
        calculator.add(value);
    }

    @Override
    public void undo() {
        calculator.restore(previousState);
    }
}

File: src/command/AddCommand.ts
import type { Command } from "./Command";
import { Calculator } from "../memento/Calculator";
import { CalculatorMemento } from "../memento/CalculatorMemento";

export class AddCommand implements Command {
    private calculator: Calculator;
    private value: number;
    private previousState?: CalculatorMemento;

    constructor(calculator: Calculator, value: number) {
        this.calculator = calculator;
        this.value = value;
    }

    execute(): void {
        this.previousState = this.calculator.save();
        this.calculator.add(this.value);
    }

    undo(): void {
        if (this.previousState) {
            this.calculator.restore(this.previousState);
        }
    }
}